Rubberduck.Refactorings/IntroduceParameter/IntroduceParameterRefactoring.cs

```diff
@@ -129,49 +134,37 @@ private bool PromptIfMethodImplementsInterface(Declaration targetVariable)
             var message = string.Format(RubberduckUI.IntroduceParameter_PromptIfTargetIsInterface,
                 functionDeclaration.IdentifierName, interfaceImplementation.ComponentName,
                 interfaceImplementation.IdentifierName);
+
             return _messageBox.Question(message, RubberduckUI.IntroduceParameter_Caption);
         }
 
         private void UpdateSignature(Declaration targetVariable)
         {
-            var functionDeclaration = _declarations.FindTarget(targetVariable.QualifiedSelection, ValidDeclarationTypes);
+            var functionDeclaration = (ModuleBodyElementDeclaration)_declarations.FindTarget(targetVariable.QualifiedSelection, ValidDeclarationTypes);
```

Comment:
is the hard-cast appropriate here? Maybe comment an explanation why this will always work

Reply:
All the members of `ValidDeclarationTypes` inherit from `ModuleBodyElementDeclaration`, so the cast will always succeed.